### PR TITLE
API 11591 Observation LastUpdated Record Search

### PR DESCRIPTION
**Pull request by that-guy-is-it**

### CODE REVIEW

File: data-query-tests/src/main/java/gov/va/api/health/dataquery/tests/SystemDefinitions.java [hunk 1]
```diff
@@ -207,6 +207,7 @@ private static Observations observations() {
         .loinc2("777-3")
         .badLoinc("99999-9")
         .onDate("2015-04-15")
+        .lastUpdated("ge2008-01-01")
```
Comment (jblefkove-va) [inline data-query-tests/src/main/java/gov/va/api/health/dataquery/tests/SystemDefinitions.java:210]: There is already `lastUpdated` defined for appointments. Adding a separate value for each resource does not seem like the correct approach; can we refactor the appointment `lastUpdated` to a shared location?

File: data-query/src/main/java/gov/va/api/health/dataquery/service/controller/observation/Dstu2ObservationController.java [hunk 2]
```diff
@@ -257,4 +257,28 @@ void replaceReferences(Collection<DatamartObservation> resources) {
             .build(),
         entitiesPage);
   }
+
+  /** Search by Last Updated. */
+  @GetMapping(params = {"_lastupdated"})
```
Comment (jblefkove-va) [inline data-query/src/main/java/gov/va/api/health/dataquery/service/controller/observation/Dstu2ObservationController.java:262]: shouldn't this be `_lastUpdated`?

Comment (that-guy-is-it) [reply data-query/src/main/java/gov/va/api/health/dataquery/service/controller/observation/Dstu2ObservationController.java:262]: Yes, typo, sorry. Will fix on next push 👍

File: data-query/src/main/java/gov/va/api/health/dataquery/service/controller/observation/Dstu2ObservationController.java [hunk 3]
```diff
@@ -257,4 +257,28 @@ void replaceReferences(Collection<DatamartObservation> resources) {
...
+    ObservationRepository.PatientAndDateSpecification spec =
+        ObservationRepository.PatientAndDateSpecification.builder()
+            .patient(cdwPatient)
+            .dates(lastUpdated)
+            .build();
+    Page<ObservationEntity> entitiesPage = repository.findAll(spec, page(page, count));
+    return bundle(
+        Parameters.builder()
+            .add("patient", patient)
+            .addAll("_lastUpdated", lastUpdated)
```
Comment (taylorrassmannlibertyits) [inline data-query/src/main/java/gov/va/api/health/dataquery/service/controller/observation/Dstu2ObservationController.java:278]: This makes the assumption that lastUpdated is not null. Is this handled by `@Valid`?

Comment (jblefkove-va) [reply data-query/src/main/java/gov/va/api/health/dataquery/service/controller/observation/Dstu2ObservationController.java:278]: I think Spring handles it, we do the same thing for parameter `date` elsewhere in this file

**Comment by jblefkove-va** [comment]
`Dstu2MetadataController` and `R4MetadataController` also need to be updated

File: data-query/src/main/java/gov/va/api/health/dataquery/service/controller/observation/Dstu2ObservationTransformer.java [hunk 4]
```diff
@@ -358,6 +368,7 @@ Observation toFhir() {
         Observation.builder()
             .resourceType("Observation")
```
Comment (jblefkove-va) [inline data-query/src/main/java/gov/va/api/health/dataquery/service/controller/observation/Dstu2ObservationTransformer.java:369]: while we're in here... remove this redundant `resourceType` here and in the r4 transformer

File: data-query/src/main/java/gov/va/api/health/dataquery/service/controller/observation/Dstu2ObservationController.java [hunk 5]
```diff
@@ -257,4 +257,28 @@ void replaceReferences(Collection<DatamartObservation> resources) {
...
+  public Observation.Bundle searchByPatientAndLastUpdated(
+      @RequestParam("patient") String patient,
+      @RequestParam("_lastUpdated") @Valid @DateTimeParameter @Size(max = 2) String[] lastUpdated,
+      @RequestParam(value = "page", defaultValue = "1") @Min(1) int page,
+      @CountParameter @Min(0) int count) {
+    String cdwPatient = witnessProtection.toCdwId(patient);
+    ObservationRepository.PatientAndDateSpecification spec =
+        ObservationRepository.PatientAndDateSpecification.builder()
+            .patient(cdwPatient)
+            .dates(lastUpdated)
```
Comment (jblefkove-va) [inline data-query/src/main/java/gov/va/api/health/dataquery/service/controller/observation/Dstu2ObservationController.java:272]: this is searching on the wrong date

**Review by jblefkove-va — APPROVED** [review]
:shipit:

**Pull request merged**